FrontEnd/components/Auth/SignupForm.vue

```diff
@@ -1,54 +1,106 @@
-            <div class="flex justify-center items-center">
-                <p>already have an account!</p> <br>
-                <button @click="handleLogin" class="text-blue-500 underline ml-2 text-lg">Login</button>
-            </div>
+      <div class="md:pt-12 space-y-2">
+        <div class="flex space-x-1">
+          <UIInput 
+            id="name" 
+            v-model="name" 
+            label="Name" 
+            placeholder="@name" 
+            :error="errors.name"
+          />
+          <UIInput 
+            id="username" 
+            v-model="username" 
+            label="Username" 
+            placeholder="@username" 
+            :error="errors.username"
+          />
         </div>
+        <UIInput 
+          id="email" 
+          v-model="email" 
+          label="E-mail" 
+          placeholder="@e-mail" 
+          :error="errors.email"
+        />
+        <UIInput 
+          id="password" 
+          v-model="password" 
+          label="Password" 
+          placeholder="Enter your password" 
+          type="password"
+          :error="errors.password"
+        />
+        <UIInput 
+          id="repassword" 
+          v-model="repassword" 
+          label="Re-password" 
+          placeholder="Re-enter your password" 
+          type="password"
+          :error="errors.repassword"
+        />
+  
+        <div>
+          <UIButtonPost 
+            liquid 
+            :disabled="isButtonDisabled" 
+            @click="handleSignUp"
+          >
+            <span>Signup</span>
+          </UIButtonPost>
+        </div>
+        <p class="text-center">or</p>
+        <div class="flex justify-center items-center">
+          <p>Already have an account!</p>
+          <button @click="handleLogin" class="text-blue-500 underline ml-2 text-lg">Login</button>
+        </div>
+      </div>
     </div>
-</template>
-
-<script setup>
-// const signup = ref(false)
-const { signup } = useAuth();
-
-const data = reactive({
-    name : '',
-    password :'',
-    email: '',
-    username : ''
-})
-
-const handleSignUp = async()=>{
-    // const credentials = { email: email.value, password: password.value };
-   try {
-    console.log(data);
-    await signup(data);
-   } catch (error) {
-    console.log(error);
-    
-   }
-}
-
-function handleLogin(){
-    navigateTo('/login')
-}
-
-const isButtonDisabled = computed(()=>{
-    return !(data.username && data.password && data.email && data.name)
-})
-
-</script>
\ No newline at end of file
+  </template>
+  
+  <script setup>
+  import { useForm, useField } from 'vee-validate';
+  import * as yup from 'yup';
+  import { useAuth } from '~/composables/useAuth'; 
+  
+  const { signup } = useAuth();
+  
+  const schema = yup.object({
+    name: yup.string().required('Name is required'),
+    username: yup.string().required('Username is required'),
+    email: yup.string().email('Invalid email address').required('Email is required'),
```

Comment:
valid email check missing

Reply:
Updated the email validation schema as suggested. Please review the latest commit